src/components/Pages/RequestPage/RequestPage.tsx

```diff
@@ -73,49 +74,37 @@ export const RequestPage = () => {
   }, [requestId])
 
   useEffect(() => {
+    if (isConnecting) return
+
+    if (!account || !provider || !providerType) {
+      console.log("The user isn't logged in, redirecting to the log in page")
+      toLoginPage()
+      return
+    }
+
     const loadRequest = async () => {
       const timeTheSiteStartedLoading = Date.now()
-      let connectionData: Awaited<ReturnType<typeof getCurrentConnectionData>> | null = null
-
-      try {
-        // Try to re-stablish connection with the wallet.
-        connectionData = await getCurrentConnectionData()
-        // Goes to the login page if no account is returned in the data.
-        if (!connectionData) {
-          throw new Error('No account connected')
-        }
-        const provider = await connection.getProvider()
-        setIsUserUsingWeb2Wallet(!!provider.isMagic)
-        providerRef.current = new ethers.BrowserProvider(provider)
-
-        const profile = await fetchProfile(connectionData.account)
-
-        // `alternative` has its own set up
-        if (!targetConfig.skipSetup) {
-          // Goes to the setup page if the connected account does not have a profile yet.
-          if (!profile) {
-            console.log("There's no profile but the user is logged in, going to setup page")
-            toSetupPage()
-            return
-          }
+      browserProvider.current = new ethers.BrowserProvider(provider)
+      const profile = await fetchProfile(account)
+
+      // `alternative` has its own set up
+      if (!targetConfig.skipSetup) {
+        // Goes to the setup page if the connected account does not have a profile yet.
+        if (!profile) {
+          console.log("There's no profile but the user is logged in, going to setup page")
+          toSetupPage()
+          return
```

Comment:
what do you think about avoid nested code?
```suggestion
      if (!targetConfig.skipSetup && !profile) {
          console.log("There's no profile but the user is logged in, going to setup page")
          toSetupPage()
          return
```

Reply:
Done!